aidies-project/client/src/components/CampaignDetail.js

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useParams, useHistory } from 'react-router-dom';
+
+import { getOneCampaign, bid, setLocation } from '../store/campaign';
+import { DetermineBid, DetermineTimeRemaining } from './CampaignUtils';
+import { makeStyles } from '@material-ui/core/styles';
+import { Card, Typography, Button, Menu, MenuItem} from '@material-ui/core';
+import '../css/campaign.css';
+
+
+const useStyles = makeStyles({
+    details: {
+        display: 'flex',
+        flexDirection: 'column',
+    },
+    bullet: {
+        display: 'inline-block',
+        margin: '0 2px',
+        transform: 'scale(0.8)',
+    },
+    title: {
+        fontSize: 14,
+    },
+    pos: {
+        marginBottom: 12,
+    },
+    button: {
+        backgroundColor: '#222',
+        color: '#fff',
+    },
+    '&$focus': {
+        backgroundColor: 'green',
+    },
+    color: {
+        color: '#222',
+        
+    },
+    delete:{
+        color: '#FF0000'
+    }
+});
+const CampaignDetail = ({ needLogin}) => {
```

Comment:
I looks like there's a lot of repetition within the JSX of your `CampainCreate` and `CampaignDetail` components! You should refactor so that the shared JSX is rendered with a reusable component. For example, I think the code below is used it both components:

```js
                    <Typography variant="h3" component="h2">
                        {campaign.name}
                    </Typography>
                    <Typography variant="subtitle2" component="h2">
                        By {campaign.User.firstName} {campaign.User.lastName} for {campaign.Charity.name}
                    </Typography>
                    <div className="detail-bid time-left">
                        <div className="detail-current-bid">
                            <Typography variant="h4" component="h2">
                                <DetermineBid campaign={campaign} />
                            </Typography>
                            <Button onClick={handleBid} size="large" variant="contained" className={classes.button}>
                                Bid $5
                            </Button>
                        </div>
                        <Typography variant="h6" component="h2" color="secondary" className="detail-remaining">
                            <DetermineTimeRemaining closingDate={campaign.closingDate} createdAt={campaign.createdAt} />
                        </Typography>
                    </div>
                    <div className="other-info">
                        <div className="category">{campaign.Category.name}</div>
                        <div className="location">
                            <div className="nav-icon"><i className="fas fa-map-marker-alt"></i></div>
                            <div>{campaign.User.location}</div>
                        </div>
                    </div>
                    <div className="information-tabs">
                        {campaign.story}
                    </div>
```

Reply:
Are you sure? My `CampaignCreate` has a bunch of input fields, while `CampaignDetail` is only displaying information

Reply:
Oh - I may have been mistaken about the details! It looked like you had some similar code between both components!